### PR TITLE
admin-gates: Add ack-4.11-kube-1.25-api-removals-in-4.12 gate

### PR DESCRIPTION
**Pull request by LalatenduMohanty**
Signed-off-by: Lalatendu Mohanty <lmohanty@redhat.com>

### CODE REVIEW

File: install/0000_00_cluster-version-operator_01_admingate_configmap.yaml [hunk 1]
```diff
@@ -1,9 +1,14 @@
 apiVersion: v1
 kind: ConfigMap
+data:
+  ack-4.11-kube-1.24-api-removals-in-4.12: |
```
Comment (bergerhoffer) [inline install/0000_00_cluster-version-operator_01_admingate_configmap.yaml:4]: @LalatenduMohanty Shouldn't this be Kube 1.25 throughout?

Comment (wking) [reply install/0000_00_cluster-version-operator_01_admingate_configmap.yaml:4]: I agree, should be 1.25, e.g. see openshift/cluster-kube-apiserver-operator#1332 adjusting the 4.11 `APIRemovedInNextReleaseInUse` to complain about removed-in-1.25 APIs.

File: install/0000_00_cluster-version-operator_01_admingate_configmap.yaml [hunk 2]
```diff
@@ -1,9 +1,14 @@
...
+      Kubernetes 1.24 and therefore OpenShift 4.12 remove several APIs which require admin consideration. Please see
+      the knowledge article https://access.redhat.com/articles/6955381 for details and instructions.
 metadata:
   name: admin-gates
   namespace: openshift-config-managed
   annotations:
     kubernetes.io/description: Define update gates for the administrator to acknowledge in the admin-acks ConfigMap in the openshift-config namespace.
     include.release.openshift.io/ibm-cloud-managed: "true"
     include.release.openshift.io/self-managed-high-availability: "true"
+    include.release.openshift.io/single-node-developer: "true"
```
Comment (wking) [inline install/0000_00_cluster-version-operator_01_admingate_configmap.yaml:14]: can we drop this?  It's dead code as I described in #685.

File: install/0000_00_cluster-version-operator_01_admingate_configmap.yaml (hunk 1 above)
Comment (LalatenduMohanty) [reply install/0000_00_cluster-version-operator_01_admingate_configmap.yaml]: Fixed in https://github.com/openshift/cluster-version-operator/commit/2d66c169658a72e1c54272646d352fc10f378b6f

File: install/0000_00_cluster-version-operator_01_admingate_configmap.yaml (hunk 2 above)
Comment (LalatenduMohanty) [reply install/0000_00_cluster-version-operator_01_admingate_configmap.yaml]: Fixed in https://github.com/openshift/cluster-version-operator/commit/2d66c169658a72e1c54272646d352fc10f378b6f

**Comment by wking** [comment]
/retitle WIP: admin-gates: Add ack-4.11-kube-1.25-api-removals-in-4.12 gate

**Review by wking — APPROVED** [review]
/lgtm

**Comment by wking** [comment]
[Storage and `Watchdog` issues][1] are unrelated.

/override ci/prow/e2e-agnostic-upgrade

[1]: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cluster-version-operator/772/pull-ci-openshift-cluster-version-operator-master-e2e-agnostic-upgrade/1521943921343598592

**Pull request merged**